src/storage/src/source/kafka.rs

```diff
@@ -761,7 +777,18 @@ impl SourceRender for KafkaSourceConnection {
                     let mut consumers = std::mem::take(&mut reader.partition_consumers);
                     for consumer in consumers.iter_mut() {
                         let pid = consumer.pid();
-                        while let Some(message) = consumer.get_next_message().transpose() {
+                        // We want to make sure the rest of the actions in the outer loops get
+                        // a chance to run. If rdkafka keeps pumping data at us we might find
+                        // ourselves in a situation where we keep dumping data into the
+                        // dataflow without signaling progress. For this reason we consume at most
+                        // 10k messages from each partition and go around the loop.
+                        let mut partition_exhausted = false;
+                        for _ in 0..10_000 {
```

Comment:
This is something that I discovered during testing that affects ingestions even when we don't use the relock_to_latest mode. Since we poll the consumer on every loop it's possible that `librdkafka` always has messages ready for us when timely schedules the operator again which means that we might not process the other things that happen in the outer loop, or ingest a particular partition until exhaustion before getting any data in from the other ones.